registration/include/pcl/registration/ndt.h

```diff
@@ -457,6 +457,16 @@ namespace pcl
       /** \brief The second order derivative of the transformation of a point w.r.t. the transform vector, \f$ H_E \f$ in Equation 6.20 [Magnusson 2009]. */
       Eigen::Matrix<double, 18, 6> point_hessian_;
 
+      /** \brief Will keep a copy of the Hessian \f$ H \f$ in Equation 6.13 [Magnusson 2009]. */
+      Eigen::Matrix<double, 6, 6> hessian_;
+
+    public:
+      /** \brief Return the hessian matrix */
+      Eigen::Matrix<double, 6, 6> getHessian()
```

Comment:
```suggestion
      Eigen::Matrix<double, 6, 6> getHessian() const
```

Reply:
Why not `const Eigen::Matrix<double, 6, 6>& getHessian() const`?

Reply:
Yep, this is also possible, gives the user more control (to avoid copy) and more responsibility (this reference will become dangling after next computation). The object is tiny so copy is cheap, so my personal preference is for safety, but up to the author ultimately. Or do we have a guideline on this @SergioRAgostinho?

Reply:
This should not be a dangling reference as long as the `NormalDistributionsTransform` exists. `hessian_` will always have the same place in memory, despite the content of `hessian_` changing with each iteration. Also a user only gets a reference if it explicitly asks for one. Most casual users will do

```cpp
Eigen::Matrix<double, 6, 6> hessian = ndt.getHessian();
```

and this will trigger the copy. I have to assume that the user which requests a const reference, knows what he/she is doing. Therefore I agree that  method should output a const ref.

Reply:
You are right, I agree with all points.